### ISSUE TITLE
error MSB1006: Property is not valid.

### ISSUE THREAD

**Issue by bmherold** [stevewillcock/grunt-msbuild#31]
Updating from 0.1.x to latest (0.3.4) yields the following error upon build:

```
Building Migrate.msbuild
Using MSBuild at:C:\Program Files (x86)\MSBuild\12.0\Bin\MSBuild.exe
Using maxcpucount: 4
Using cmd: C:\Program Files (x86)\MSBuild\12.0\Bin\MSBuild.exe
Using args: [ 'Migrate.msbuild',
  '/target:Migrate',
  '/verbosity:minimal',
  '/nologo',
  '/maxcpucount:4',
  '/property:Configuration=local',
  '/property:DryRun=False',
  '/property:Verbose=False',
  '/property:RollbackSteps=1',
  '/property:ConnectionString=Data Source=COMPUTERNAME;Initial Catalog=MYDATABASE;Integrated Security=True;' ]
MSBUILD : error MSB1005: Specify a property and its value.
Switch: /property:ConnectionString=Data Source=COMPUTERNAME;Initial Catalog=MYDATABASE;Integrated Security=True;

For switch syntax, type "MSBuild /help"
close received - code:  false
MSBuild failed with code: 1Migrate.msbuild
Warning: MSBuild exited with a failure code: 1 Use --force to continue.
```

My grunt config is using the following:

```
msbuild: {
  options: {
    projectConfiguration: 'Dev',
    targets: ['Clean', 'Rebuild'],
    maxCpuCount: 4,
    verbosity: 'minimal',
    stdout: true,
    buildParameters: {
      WarningLevel: 2,
      DeployOnBuild: false
    },
  },  
  migrate: {
    src: ['Migrate.msbuild'],
    options: {
        targets: ['Migrate'],
        buildParameters: {
          DryRun: 'False',
          Verbose: 'False',
          RollbackSteps: '1',
          ConnectionString: 'Data Source=COMPUTERNAME;Initial Catalog=MYDATABASE;Integrated Security=True;'
        },
        verbosity: 'minimal',
    }   
  }
}
```

It seems that the ConnectionString property is the culprit - if removed the build works. Searching the internet seems to indicate the need to escape or quote property values with a semicolon properly. I've tried a number of permutations of quotes to no avail. Any thoughts?

**Comment by stevewillcock** [comment]
Adding the buildParameters section that you have there to an msbuild config here led to the same error. I was able to make the build work by replacing the semicolons in the connection string with the escape chars %3B - i.e. 

    buildParameters : {
            WarningLevel : 2,
            DryRun : 'False',
            Verbose : 'False',
            RollbackSteps : '1',
            ConnectionString :
                'Data Source=COMPUTERNAME%3BInitial Catalog=MYDATABASE%3BIntegrated Security=True%3B'
          },

The special characters for MSBuild are detailed here: https://msdn.microsoft.com/en-us/library/ms228186.aspx and here: https://msdn.microsoft.com/en-us/library/bb383819.aspx 
I'm not actually using the connection string parameter in my build though, so I can't confirm that this is working as expected (the build completes but I can't tell whether the connection string parameter is properly applied to the build). Indeed you may have already tried this and found that it didn't work. If you could let me know whether this works for you that would be great.

Thanks

**Comment by bmherold** [comment]
Hmm, I think we're closer? Is there an expected/allowed character I can wrap around the Data Source property value? I escaped the semicolons as per your article and example but I'm getting a different error. I ran it with grunt-msbuild 0.1.x sucessfully, but changing over to 0.3.4 gave another error. Linking gists for both if that helps. 

0.1.x working: https://gist.github.com/bmherold/226907e1256e7172e3b4
0.3.4 error: https://gist.github.com/bmherold/db11fdbd9be07288385d

Let me know what else I can provide - thanks!
